src/test/groovy/edu/ie3/datamodel/io/source/influxdb/InfluxDbWeatherSourceIconIT.groovy

```
@@ -118,31 +119,42 @@ class InfluxDbWeatherSourceIconIT extends Specification implements WeatherSource
     given:
     def validCoordinate = IconWeatherTestData.COORDINATE_67775
     def invalidCoordinate = GeoUtils.buildPoint(7d, 48d)
     def time = IconWeatherTestData.TIME_15H
     def timeInterval = new ClosedInterval(IconWeatherTestData.TIME_15H , IconWeatherTestData.TIME_17H)
-    def emptyTimeSeries = new IndividualTimeSeries(UUID.randomUUID(), Collections.emptySet())
     def timeseries67775 = new IndividualTimeSeries(null,
         [
           new TimeBasedValue(IconWeatherTestData.TIME_15H, IconWeatherTestData.WEATHER_VALUE_67775_15H),
           new TimeBasedValue(IconWeatherTestData.TIME_16H, IconWeatherTestData.WEATHER_VALUE_67775_16H),
           new TimeBasedValue(IconWeatherTestData.TIME_17H, IconWeatherTestData.WEATHER_VALUE_67775_17H)
         ] as Set<TimeBasedValue>)
 
-    when:
-    def coordinateAtDate = source.getWeather(time, invalidCoordinate)
-    def coordinateInInterval = source.getWeather(timeInterval, invalidCoordinate)
+    when: "requesting weather for an invalid coordinate at a specific date"
+    source.getWeather(time, invalidCoordinate)
+
+    then: "NoDataException is thrown"
+    def ex1 = thrown(NoDataException)
+    ex1.message.contains("No coordinate ID found for the given point")
+    ex1.message.contains(invalidCoordinate.toString())
+
+    when: "requesting weather for an invalid coordinate in a time interval"
+    source.getWeather(timeInterval, invalidCoordinate)
+
+    then: "NoDataException is thrown"
+    def ex2 = thrown(NoDataException)
+    ex2.message.contains("No data for given coordinates")
+    ex2.message.contains(invalidCoordinate.toString())
+
+    when: "requesting weather for mixed valid and invalid coordinates"
     def coordinatesToTimeSeries = source.getWeather(timeInterval, [
       validCoordinate,
       invalidCoordinate
     ])
 
-    then:
-    coordinateAtDate == Optional.empty()
-    equalsIgnoreUUID(coordinateInInterval, emptyTimeSeries)
+    then: "only valid coordinate data is returned"
```

Do we want this? I think if we would throw an exception when requesting some invalid coordinate, then it should still be thrown if it comes along with valid ones.

---

I dont know if Im wrong at this point, but if I remember and understood correctly the approaches for handling this were not consistent through the various WeatherSources. Still have to check that, yesterday was kind of a long coding session :D
But I will change everything to strict failure if no or damaged data is found. I think this would be the safest approach and saving the user time by following the fail fast philosophy.